### PR TITLE
Added Scheduler Module

### PR DESCRIPTION
**Pull request by therealsgopi**
Added a README for Scheduler module and also added the code for some basic scheduling (with minimum features) considering my own sample data.
Let me know your views on this update

### CODE REVIEW

**Review by PratikGarai — CHANGES_REQUESTED** [review]
Add reschedule / push back feature so single interviews can be preponed or postponed.

**Comment by therealsgopi** [comment]
Noted.
Will get back to you soon.

File: Scheduler module/scheduler.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+# collecting basic info of interview schedule
+dur = int(input("Enter duration of 1 interview (in minutes): "))
+brk = int(input("Enter duration of break after 1 interview (in minutes): "))
+start_date = list(map(int,input("Enter starting day of interviews (DD:MM:YYYY): ").split(":")))
+start_time = list(map(int,input("Enter starting time of interviews everyday (HH:MM) (24 Hr Format): ").split(":")))
+end_time = list(map(int,input("Enter ending time of interviews everyday (HH:MM) (24 Hr Format): ").split(":")))
+now = datetime.datetime.now()
+
+schedule()
```
Comment (PratikGarai) [inline Scheduler module/scheduler.py:59]: Pass dur, brk, start_date, start_time, end_time as arguments to schedule method

File: Scheduler module/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# sorting raw data based on year of candidate - ascending
+candidates = dict(sorted(candidates.items(), key=lambda item: item[1]["year"]))
+
+# collecting basic info of interview schedule
+dur = int(input("Enter duration of 1 interview (in minutes): "))
+brk = int(input("Enter duration of break after 1 interview (in minutes): "))
+start_date = list(map(int,input("Enter starting day of interviews (DD:MM:YYYY): ").split(":")))
+start_time = list(map(int,input("Enter starting time of interviews everyday (HH:MM) (24 Hr Format): ").split(":")))
+end_time = list(map(int,input("Enter ending time of interviews everyday (HH:MM) (24 Hr Format): ").split(":")))
+now = datetime.datetime.now()
```
Comment (PratikGarai) [inline Scheduler module/scheduler.py:57]: calculate this inside the schedule() method

**Review by PratikGarai — CHANGES_REQUESTED** [review]

File: Scheduler module/scheduler.py (hunk 1 above)
Comment (therealsgopi) [reply Scheduler module/scheduler.py:59]: Got it, will make it modular

File: Scheduler module/scheduler.py (hunk 2 above)
Comment (therealsgopi) [reply Scheduler module/scheduler.py:57]: This is actually not even required, I was just experimenting something and forgot to remove it.

Comment (PratikGarai) [reply Scheduler module/scheduler.py:57]: okay

File: Scheduler module/scheduler.py (hunk 1 above)
Comment (therealsgopi) [reply Scheduler module/scheduler.py]: Done.

File: Scheduler module/api.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from fastapi import FastAPI
+import scheduler as sch
+
+scheduler = FastAPI()
+
+@scheduler.get("/schedule/duration={dur}_break={brk}_start-date={st_dt}_start-time={st_tm}_end-time={end_tm}")
```
Comment (PratikGarai) [inline Scheduler module/api.py:6]: Define a proper request response model using Pydantic

**Review by PratikGarai — CHANGES_REQUESTED** [review]

File: Scheduler module/scheduler.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+# raw data got from backend
+candidates = {"20bce1800":{"name":"Raghu", "course":"CSE", "year":2020},
+            "20bce1802":{"name":"Arvind", "course":"MECH", "year":2020},
+            "20bce1803":{"name":"Vaishnavi", "course":"CSE", "year":2021},
+            "20bce1804":{"name":"Gopal", "course":"ECE", "year":2022},
+            "20bce1805":{"name":"Shivam", "course":"EEE", "year":2021}}
+
+# sorting raw data based on year of candidate - ascending
+candidates = dict(sorted(candidates.items(), key=lambda item: item[1]["year"]))
+# dont think this is necessary as I will have to just scheduler in the order I get (in case rescheduled to create void etc)
```
Comment (PratikGarai) [inline Scheduler module/scheduler.py:56]: keep comments less personalized, make them as generic as possible

**Review by PratikGarai — CHANGES_REQUESTED** [review]

File: Scheduler module/scheduler.py (hunk 4 above)
Comment (therealsgopi) [reply Scheduler module/scheduler.py:56]: Okay bhaiya, noted. Will keep this in mind.

File: Scheduler module/api.py (hunk 3 above)
Comment (therealsgopi) [reply Scheduler module/api.py]: Done bhaiya, please check

**Review by PratikGarai — APPROVED** [review]

**Pull request merged**